app/components/sy-durationpicker/component.js

```diff
@@ -70,6 +72,26 @@ export default SyTimepickerComponent.extend({
     return value ? formatDuration(value, false) : ''
   },
 
+  /**
+   * Get the duration for given hours and minutes
+   *
+   * If one of the given numbers is negative, both have to be.
+   *
+   * @method _getDurationFromHoursAndMinutes
+   * @param {Number} h The hours of the wanted duration
+   * @param {Number} m The minutes of the wanted duration
+   * @return {moment.duration} The resulted duration
+   * @private
+   */
+  _getDurationFromHoursAndMinutes(h, m) {
+    let negative = [h, m].some(n => n < 0)
+
+    h = abs(h) * (negative ? -1 : 1)
+    m = abs(m) * (negative ? -1 : 1)
+
+    return moment.duration({ h, m })
```

Comment:
In the test, you're trying a negative value for `h` and a positive one for `m`. Is the only case that makes sense? If yes, i'd also implement it explicitly like that, and not generically like above, because like this you have 
```
h=1, m=-30 => -1:30
h=-1, m=-30 => -1:30
```
which might be a little confusing.

Reply:
I see your point. Maybe we should extract negative as a parameter and pass the hours and minutes as positive numbers. What do you think?

Reply:
ATM we extract hours and minutes from a string by `let [h, m] = str.split(':').map(n => parseInt(n))` which results in only negative hours but not minutes. But we also have the use case where we pass negative minutes to add (when using the arrow keys). So extracting a parameter whether the duration is negative or not might be the best solution.

Reply:
As discussed: extra parameter is fine!